docs/php/activities.md

```diff
@@ -3,3 +3,309 @@ id: activities
+            fopen($this->localDirectory . $localName, 'rb+')
+        );
+    }
+
+    public function download(
+        string $bucketName,
+        string $remoteName,
+        string $localName
+    ): void
+    {
+        $this->s3Client->downloadObject(
+            $bucketName,
+            $remoteName,
+            fopen($this->localDirectory .$localName, 'wb+')
+        );
+    }
+
+    public function processFile(string $localName): string
+    {
+        // Implementation omitted for brevity.
+        return compressFile($this->localDirectory . $localName);
+    }
+
+    public function deleteLocalFile(string $fileName): void
+    {
+        unlink($this->localDirectory . $fileName);
+    }
+}
+```
+
+### Accessing Activity Info
+
+The [Activity](https://github.com/temporalio/sdk-php/blob/master/src/Activity.php) class provides static getters to access information about the Workflow that invoked it.
+Note that this information is stored in a thread local variable.
+Therefore, calls to Activity accessors succeed only in the process that invoked the Activity function.
+
+```php
+use Temporal\Activity;
+
+class FileProcessingActivitiesImpl implements FileProcessingActivities
+{
+    // ...
+
+    public function download(
+        string $bucketName,
+        string $remoteName,
+        string $localName
+    ): void
+    {
+        $this->logger->info("namespace=" +  Activity.getNamespace());
+
+        $info = Activity::getInfo();
+
+        $this->logger->info("workflowId=" + $info->workflowExecution->getID());
+        $this->logger->info("runId=" + $info->workflowExecution->getRunID());
+        $this->logger->info("activityId=" + $info->id);
+        $this->logger->info("activityDeadline=" + $info->deadline);
+
+        // ...
+    }
+
+    // ...
+}
+```
+
+### Activity Heart Beating
+
+Some Activities are long-running.
+To react to a crash quickly, use a heartbeat mechanism.
+`Activity::heartbeat()` lets the Temporal service know that the Activity is still alive.
+You can piggyback `details` on an Activity heartbeat.
+If an Activity times out, the last value of `details` is included in the `TimeoutFailure` delivered to a Workflow.
+Then the Workflow can pass the details to the next Activity invocation.
+This acts as a periodic checkpoint mechanism for the progress of an Activity.
+
+```php
+use Temporal\Activity;
+
+class FileProcessingActivitiesImpl implements FileProcessingActivities
+{
+    // ...
+    public function download(
+        string $bucketName,
+        string $remoteName,
+        string $localName
+    ): void
+    {
+        $this->dowloader->downloadWithProgress(
+            $bucketName,  
+            $remoteName,  
+            $localName,
+            // on progress
+            function ($progress) {
+                Activity::heartbeat($progress);
+            }
+        );
+
+        Activity::heartbeat(100); // download complete
+
+        // ...
+    }
+
+    // ...
+}
+```
+
+## Calling Activities
```

Comment:
I think calling activities section belongs to workflows, not to the activities part of the documentation.